### PR TITLE
impr: use object as set

### PR DESCRIPTION
**Pull request by JanLewDev**

### CODE REVIEW

File: packages/object/src/hashgraph/index.ts [hunk 1]
```diff
@@ -13,6 +13,29 @@ export type { Vertex, Operation };
...
+		this.set = {} as { [key in T]: boolean };
+		for (const item of iterable) {
+			this.set[item] = true;
+		}
+	}
+
+	add(item: T): void {
+		this.set[item] = true;
+	}
+
```
Comment (trungnotchung) [inline packages/object/src/hashgraph/index.ts:29]:
```suggestion
delete(item: T): void {
		delete this.set[item];
	}
```

File: packages/object/src/hashgraph/index.ts [hunk 2]
```diff
@@ -13,6 +13,29 @@ export type { Vertex, Operation };
 
 export type Hash = string;
 
+export class ObjectAsSet<T extends string | number | symbol> {
```
[the remark this answers is not in the document]
Comment (JanLewDev) [reply packages/object/src/hashgraph/index.ts:16]: under hashgraph dir?

[the remark this answers is not in the document]
Comment (JanLewDev) [reply packages/object/src/hashgraph/index.ts:16]: under hashgraph dir?

**Review by d-roak — APPROVED** [review]
lgtm

**Review by trungnotchung — APPROVED** [review]
lgtm

**Pull request merged**